[quant][fx] Move embedding ops to backend_config_dict

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75469
* #75389
* #75388
* #75387
* #75386
* #75378
* #75377
* #75318
* __->__ #75317
* #75316

Summary:
att, similar to previous PRs, this one moves dynamically quantized rnn ops
to backend_config_dict

we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
from prepare, then to convert. We can start this process after this PR

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestFXNumericSuiteCoreAPIs

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35423789](https://our.internmc.facebook.com/intern/diff/D35423789)

jerryzh168 committed:
[quant][fx] Move embedding ops to backend_config_dict

Summary:
att, similar to previous PRs, this one moves dynamically quantized rnn ops
to backend_config_dict

we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
from prepare, then to convert. We can start this process after this PR

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestFXNumericSuiteCoreAPIs

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423789).

vkuzo commented:
> we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
from prepare, then to convert. We can start this process after this PR

thanks for the context, could we document the code with what is temporary?

jerryzh168 commented:
> > we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
> > everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
> > from prepare, then to convert. We can start this process after this PR
> 
> thanks for the context, could we document the code with what is temporary?

yeah we have that in the code

jerryzh168 commented on torch/ao/quantization/fx/backend_config/quantize_handler.py:
```diff
@@ -57,5 +59,9 @@ def get_activation_ctr(
                     return self.overwrite_output_observer
             return qconfig.activation
 
+        # This is temporary, and will be removed soon
```
this function, and also a key in backend_config_dict

jerryzh168 committed:
Update on "[quant][fx] Move embedding ops to backend_config_dict"


Summary:
att, similar to previous PRs, this one moves dynamically quantized rnn ops
to backend_config_dict

we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
from prepare, then to convert. We can start this process after this PR

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestFXNumericSuiteCoreAPIs

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35423789](https://our.internmc.facebook.com/intern/diff/D35423789)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx] Move embedding ops to backend_config_dict"


Summary:
att, similar to previous PRs, this one moves dynamically quantized rnn ops
to backend_config_dict

we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
from prepare, then to convert. We can start this process after this PR

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestFXNumericSuiteCoreAPIs

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35423789](https://our.internmc.facebook.com/intern/diff/D35423789)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423789).

jerryzh168 committed:
Update on "[quant][fx] Move embedding ops to backend_config_dict"


Summary:
att, similar to previous PRs, this one moves dynamically quantized rnn ops
to backend_config_dict

we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
from prepare, then to convert. We can start this process after this PR

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestFXNumericSuiteCoreAPIs

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35423789](https://our.internmc.facebook.com/intern/diff/D35423789)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx] Move embedding ops to backend_config_dict"


Summary:
att, similar to previous PRs, this one moves dynamically quantized rnn ops
to backend_config_dict

we have some temporary configs in backend_config_dict, but it will be removed soon, we want to migrate
everything to backend_config_dict so that we can enable this path for all the code in the code base, starting
from prepare, then to convert. We can start this process after this PR

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestFXNumericSuiteCoreAPIs

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35423789](https://our.internmc.facebook.com/intern/diff/D35423789)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423789).
